src/fem/fe_values.cc

```diff
@@ -505,8 +514,20 @@ FEValues<dim,spacedim>::FEValues(Mapping<dim,spacedim> &_mapping,
          UpdateFlags _flags)
 :FEValuesBase<dim, spacedim>()
 {
-    this->allocate(_mapping, _quadrature, _fe, _flags);
+    this->allocate(_mapping, _quadrature, _fe, _fe.n_dofs(), _flags);
 
+    //UpdateFlags map_update_flags = this->data.update_flags;
+    
+    if(typeid(_quadrature) == typeid(QXFEM<dim,spacedim>)){
```

Comment:
We should make this more general, either add a method "reset_quadrature" or introduce derived class where quadrature is not fixed. Some similarities to p-adaptivity.